File: gen_scripts/perl/e60.branch.pl
#!/usr/bin/env perl -w
print "Input a character: "; $keypress=getc(STDIN);

if ($keypress =~ /[[:lower:]]/) {
    print "Lowercase letter\n";
} elsif ($keypress =~ /[[:upper:]]/) {
    print "Uppercase letter\n";
} elsif ($keypress =~ /[[:digit:]]/) {
    print "Digit\n";
} else {
    print "Punctuation, whitespace, or other\n";
}

File: gen_scripts/python/e60.branch.py
#!/usr/bin/env python
import sys  # system library for standard input and output
import re   # regular expresion library
 
sys.stdout.write("Input a character: ")    # output prompt
keypress = sys.stdin.read(1)               # read one character

if re.match("[a-z]", keypress):
  print "Lowercase letter"
elif re.match("[A-Z]", keypress):
  print "Uppercase letter"
elif re.match("[0-9]", keypress):
  print "Digit"
else:
  print "Punctuation, whitespace, or other"